### PR TITLE
upgrade jsonfield for sms

### PR DESCRIPTION
**Pull request by nickpell**
upgrading our third party JSONField library from 0.5.7 to 1.0.3 (same workflow as https://github.com/dimagi/commcare-hq/pull/10432)

### CODE REVIEW

File: corehq/apps/sms/migrations/0003_add_backend_models.py [hunk 1]
```diff
@@ -27,10 +27,10 @@ class Migration(migrations.Migration):
                 ('name', models.CharField(max_length=126)),
                 ('display_name', models.CharField(max_length=126, null=True)),
                 ('description', models.TextField(null=True)),
-                ('supported_countries', json_field.fields.JSONField(default=[], help_text='Enter a valid JSON object')),
-                ('extra_fields', json_field.fields.JSONField(default={}, help_text='Enter a valid JSON object')),
+                ('supported_countries', jsonfield.fields.JSONField(default=[])),
```
Comment (emord) [inline corehq/apps/sms/migrations/0003_add_backend_models.py]: `list`?

Comment (nickpell) [inline corehq/apps/sms/migrations/0003_add_backend_models.py]: 👍

**Comment by snopoke** [comment]
:+1: once mutables defaults are fixed

**Comment by nickpell** [comment]
✅ @esoergel

**Pull request merged**